js/libs/modal.js

```diff
@@ -248,9 +248,12 @@ var $ = require('jquery');
   $(document)
     .on('show.bs.modal',  '.modal', function () { $(document.body).addClass('modal-open') })
     .on('hidden.bs.modal', '.modal', function () {
-      // $(this).removeData('bs.modal').empty()
       $(document.body).removeClass('modal-open');
+    });
 
-    })
-
+  // Focus the first input in a modal when the modal has been opened
+  $(document).on('shown.bs.modal', '.modal', function() {
+    $(':input:not(input[type=button], input[type=submit], button):visible:first', $(this)).focus();
```

Comment:
This selector is on the large side, if we are just trying to focus the first focusable input would jQuery's `:focusable` help shorten this at all?

Why are we passing `$(this)` as the context argument to the selector? Shouldn't the context already be set to `.modal` (I could be wrong here, not been in jQuery land for a while), if it's the case that context does need to be explicitly set, we could probably omit creating a jQuery object out of it and just pass in `this`? This is admittedly a really minor probablly-not-worth-doing change!

Reply:
> if we are just trying to focus the first focusable input would jQuery's :focusable help shorten this at all?

We need the first focusable element that's not a button, so wouldn't that just be changing 

`:input:not(input[type=button], input[type=submit], button)` 

to 

`:focusable:not(input[type=button], input[type=submit], button)`

which is longer?

Reply:
Right gotcha! Yep good point. I find these pseudo CSS query string really difficult to reason about at a glance, I think I personally prefer chaining jQuery's filter methods, something like 

```$(':input:visible').not('[type="button"],[type="submit"],button).first();```

That is purely my preference though, I think there's some non realisable optimisation as the query scope is reduced in each chain rather than one mega query.